File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = [1] * len(nums)
        for j, nt in enumerate(nums):
            for i, n in enumerate(nums):
                if j != i: 
                    output[j] *= n

        return output


        # go over nums skipping the ith element
        # have a variable that *= all of them and at the end of that loop
        # have a master for loop so that we basically go over the nums array twice

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for(int j = 0; j < nums.size(); j++){
            for(int i = j + 1; i < nums.size(); i++){
                if(nums[i] + nums[j] == target){
                    return {j,i};
                }
            }
        }
        return{};
    }
};
